core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -143,7 +143,7 @@
     public static final Setting<Boolean> NODE_INGEST_SETTING =
         Setting.boolSetting("node.ingest", true, Property.NodeScope);
     public static final Setting<String> NODE_NAME_SETTING = Setting.simpleString("node.name", Property.NodeScope);
-    // this sucks that folks can mistype client etc and get away with it.
+    // this sucks that folks can mistype data, master or ingest and get away with it.
```

Comment:
I wonder if we should fix that at the same go and introduce `node.attributes.${someuserattr}` and make it strict? How much easier would your live become here?